experimental/rules_python/wheelmaker.py

```diff
@@ -93,8 +94,18 @@ def add_string(self, filename, contents):
 
     def add_file(self, package_filename, real_filename):
         """Add given file to the distribution."""
+        def strip_prefix(path, prefix):
+            if path.startswith(prefix):
+                return path[len(prefix):]
+            else:
+                return path
+
+        resolved_package_filename = package_filename \
+            if self._package_root_path is None \
+            else strip_prefix(package_filename, self._package_root_path)
+
         # Always use unix path separators.
-        arcname = package_filename.replace(os.path.sep, '/')
+        arcname = resolved_package_filename.replace(os.path.sep, '/')
```

Comment:
Shouldn't this happen before strip_prefix? I'd suppose BUILD file would specify the prefix in unix style regardless of build platform, so stripping would need to use '/' separator to work reliably on windows.

Reply:
Correct! Fixed.
I incorporated all the logic into one function that returns the final `arcname` in order to avoid another variable that can be accidentally used instead of the correct one.